ext/include/opentelemetry/ext/zpages/tracez_http_server.h

```diff
@@ -0,0 +1,125 @@
+   * @param host is the host where the TraceZ webpages will be displayed, default being localhost
+   * @param port is the port where the TraceZ webpages will be displayed, default being 30000
+   */
+  TracezHttpServer(std::unique_ptr<opentelemetry::ext::zpages::TracezDataAggregator> &&aggregator,
+                   std::string host = "localhost", int port = 30000) :
+                   opentelemetry::ext::zpages::zPagesHttpServer("/tracez/get", host, port),
+                   data_aggregator_(std::move(aggregator)) {
+    InitializeTracezEndpoint(*this);
+    InitializeFileEndpoint(*this);
+  };
+
+ private:
+   /*
+   * Set the HTTP server to use the "Serve" callback to send the appropriate data when queried
+   * @param server, which should be an instance of this object
+   */
+  void InitializeTracezEndpoint(TracezHttpServer& server) {
+    server[endpoint_] = Serve;
+  }
+
+  /*
+   * Updates the stored aggregation data (aggregations_) using the data aggregator
+   */
+  void UpdateAggregations();
+  
+  /*
+   * First updates the stored aggregations, then translates that data from a C++ map to
+   * a JSON object
+   * @returns JSON object of collected spans bucket counts by name
+   */
+  json GetAggregations();
+
+  /*
+   * Using the stored aggregations, finds the span group with the right name and returns
+   * its running span data as a JSON, only grabbing the fields needed for the frontend
+   * @param name of the span group whose running data we want
+   * @returns JSON representing running span data with the passed in name
+   */
+  json GetRunningSpansJSON(const std::string& name);
+
+  /*
+   * Using the stored aggregations, finds the span group with the right name and returns
+   * its error span data as a JSON, only grabbing the fields needed for the frontend
+   * @param name of the span group whose running data we want
+   * @returns JSON representing eoor span data with the passed in name
+   */
+  json GetErrorSpansJSON(const std::string& name);
+
+  /*
+   * Using the stored aggregations, finds the span group with the right name and bucket index
+   * returning its latency span data as a JSON, only grabbing the fields needed for the frontend
+   * @param name of the span group whose latency data we want
+   * @param index of which latency bucket to grab from
+   * @returns JSON representing bucket span data with the passed in name and latency range
+   */
+  json GetLatencySpansJSON(const std::string& name, const int& latency_range_index);
+  
+  /*
+   * Sets the response object with the TraceZ aggregation data based on the request endpoint
+   * @param req is the HTTP request, which we use to figure out the response to send
+   * @param resp is the HTTP response we want to send to the frontend, either webpage or TraceZ
+   * aggregation data 
+   */
+  HTTP_SERVER_NS::HttpRequestCallback Serve{[&](HTTP_SERVER_NS::HttpRequest const& req,
```

Comment:
I remember there was discussion about this function potentially being access by multiple threads -- is that still an issue?

Reply:
If there is, I think we're not prioritizing it right now and it's outside of the intern project scope? I'm having issues getting more details so the HTTP server base class can get merged, so it will likely get added to OT as is